File: Add_Border/Add_border.js
function solution(array) {
  let solution = new Array()

  for (let i = 0; i < array.length; i++) 
    solution [i + 1] = "*" + array[i] + "*"

  let border = ""
  for(; border.length < array[0].length; border += "*"){}

  solution[0] = solution[solution.length] = border += "**"

  return solution
}

const tests = [
	["abc", "ded"],
        ["wzy**"],
        ["abcde", "fghij", "klmno", "pqrst", "uvwxy"]
]

for (const test of tests) {
  const testString = test.join(",\n ")
  const solutionString = solution(test).join(",\n ")

  console.log("Without border:\n", testString, "\nWith border:\n", solutionString, "\n\n");
}

File: Add_Border/Add_border.java
import java.util.Arrays;
public class Add_border{
	public static void main (String[] args){
            String [][] tests = {
                {"abc", "ded"},
                {"wzy**"},
                {"abcde", "fghij", "klmno", "pqrst", "uvwxy"}
            };
            for(String[] test : tests){
                var testString = Arrays.toString(test).replace(",", ",\n ");
                var solutionString= Arrays.toString(solution(test)).replace(",", ",\n ");
                System.out.format(" Without border:%n %s %n With border:%n %s %n%n%n", testString, solutionString);
            }
        }
	

	static String[] solution(String[] picture){
            String[] solution = new String[picture.length + 2];

            for (int i = 0; i < picture.length; i++)
                solution[i + 1] = "*" + picture[i] + "*";

            String border = "";
            for(; border.length() < picture[0].length() + 2; border += '*'){}

            solution [0] = solution[solution.length - 1] = border;
            return solution;
        }
}
